sabnzbd/sorting.py

```diff
@@ -797,218 +546,180 @@ def filter_files(_file, current_path):
+    """Move all files under 'workdir' into 'workdir/..'"""
+    # Determine 'folder'/..
+    workdir = os.path.abspath(os.path.normpath(workdir))
+    dest = os.path.abspath(os.path.normpath(os.path.join(workdir, "..")))
 
-        self.date_info["ep_name"], self.date_info["ep_name_two"], self.date_info["ep_name_three"] = get_descriptions(
-            self.nzo, self.match_obj, self.original_job_name
-        )
+    # Check for DVD folders and bail out if found
+    for item in os.listdir(workdir):
+        if item.lower() in ("video_ts", "audio_ts", "bdmv"):
+            return workdir, True
 
-        return True
+    for root, dirs, files in os.walk(workdir):
+        for _file in files:
+            path = os.path.join(root, _file)
+            new_path = path.replace(workdir, dest)
+            ok, new_path = move_to_path(path, new_path)
+            if not ok:
+                return dest, False
 
-    def construct_path(self):
-        """Return path reconstructed from original and sort expression"""
-        sorter = self.sort_string.replace("\\", "/")
-        mapping = []
+    cleanup_empty_directories(workdir)
+    return dest, True
 
-        if ends_in_file(sorter):
-            extension = True
-            sorter = sorter.replace(".%ext", "")
-        else:
-            extension = False
 
-        # Replace title
-        mapping.append(("%title", self.date_info["title"]))
-        mapping.append(("%.title", self.date_info["title_two"]))
-        mapping.append(("%_title", self.date_info["title_three"]))
-
-        mapping.append(("%t", self.date_info["title"]))
-        mapping.append(("%.t", self.date_info["title_two"]))
-        mapping.append(("%_t", self.date_info["title_three"]))
-
-        mapping.append(("%sn", self.date_info["ttitle"]))
-        mapping.append(("%s.n", self.date_info["ttitle_two"]))
-        mapping.append(("%s_n", self.date_info["ttitle_three"]))
-        mapping.append(("%sN", self.date_info["title"]))
-        mapping.append(("%s.N", self.date_info["title_two"]))
-        mapping.append(("%s_N", self.date_info["title_three"]))
-
-        # Replace year
-        mapping.append(("%year", self.date_info["year"]))
-        mapping.append(("%y", self.date_info["year"]))
-
-        # Replace resolution
-        mapping.append(("%r", self.date_info["resolution"]))
-
-        if self.date_info["ep_name"]:
-            mapping.append(("%desc", self.date_info["ep_name"]))
-            mapping.append(("%.desc", self.date_info["ep_name_two"]))
-            mapping.append(("%_desc", self.date_info["ep_name_three"]))
+def guess_what(name: str, sort_type: Optional[str] = None) -> MatchesDict:
+    """Guess metadata for movies or episodes from their name. The sort_type ('movie' or 'episode')
+    is passed as a hint to guessit, if given."""
+
+    if not name:
+        raise ValueError("Need a name for guessing")
+
+    # Remove any passwords from the name
+    name = scan_password(name)[0]
```

Comment:
When we get to here, the name can still include the password? That should not be the case..

Reply:
The sorting is called in more places than just postproc, and in at least some of those the password is still in the name. For example [`newsunpack.pre_queue`](https://github.com/sabnzbd/sabnzbd/blob/399de15792c878d74fedb0808d6435c620c5ac79/sabnzbd/newsunpack.py#L2337) that feeds the jobname - password included - to the `SeriesSorter` via `newsunpack.analyse_show` in order to supply the script with show info.